lib/actions/equalizer.js

```diff
@@ -1,13 +1,21 @@
 'use strict';
 
 function nightMode(player, values) {
-  const enable = values[0] === 'on';
-  return player.nightMode(enable);
+  let enable = values[0] === 'on';
+  let ret = { status: 'success' };
+  if(values[0] == "toggle") enable = ret.nightmode = !player.coordinator.state.equalizer.nightMode;
+  return player.nightMode(enable).then((response) => {
+    return (!response || response.constructor.name === 'IncomingMessage') ? ret : response;
+  });
```

Comment:
It would be much cleaner to always respond with the same json response for all calls (not only toggle).

So basically always take response, pad with the status from the player and return. The typ checking for IncomingMessage seems unnecessary, on fault it would reject the promise.

Reply:
> It would be much cleaner to always respond with the same json response for all calls (not only toggle).
> 
> So basically always take response, pad with the status from the player and return.

Easy change, will do.

> The typ checking for IncomingMessage seems unnecessary, on fault it would reject the promise.

So, I'm replicating the behaviour of [this code](https://github.com/jishi/node-sonos-http-api/blob/master/lib/sonos-http-api.js#L96-L97)... as mentioned, I'm not a nodejs developer, so really not familiar enough to know if this is required at all. I added the check so that the response is correctly passed down the chain if it's not an `IncomingMessage`, instead of blindly overwriting it. Your thoughts here would be welcome.

Reply:
Yeah, you don't need to bother with that, it covers some corner case when it returns a raw http response. You just return whatever json payload you want it to render and it will be taken care of by https://github.com/jishi/node-sonos-http-api/blame/master/lib/sonos-http-api.js#L102

Reply:
Ok, I've rebased onto master, and added a commit that removes the check, and returns state for on/off too.